curriculum/challenges/english/20-upcoming-python/learn-tree-traversal-by-building-a-binary-search-tree/655a44442b60ee5a28df8ee5.md

```diff
@@ -0,0 +1,75 @@
+---
+id: 655a44442b60ee5a28df8ee5
+title: Step 25
+challengeType: 20
+dashedName: step-25
+---
+
+# --description--
+
+If the second `if` statement is not `True`, it means that the target key is greater than or equal to the current node key.
+
+After the `if` block, return the result of calling the `_search` method with the right child of the current node.
+
+This step is taken because, in a binary search tree, if the target key is greater than the current node's key, the search continues in the right subtree.
```

Comment:
```suggestion
If the second `if` statement is not `True`, it means that the target key is greater than or equal to the current node key. 

In a binary search tree, if the target key is greater than the current node's key, the search continues in the right subtree.

After the `if` block, return the result of calling the `_search` method with the right child of the current node and the key as arguments.
```
I would suggest rearranging the last two paragraphs to explain the logic of why the search continues in the right sub-tree before writing the code.